wallet/wallet.c

```diff
@@ -3090,17 +3091,27 @@ void wallet_payment_set_failinfo(struct wallet *wallet,
 	db_exec_prepared_v2(take(stmt));
 }
 
-const struct wallet_payment **
+static int cmp_wallet_payment(const struct wallet_payment *a,
+			 		const struct wallet_payment *b,
+			 		void *unused UNUSED)
+{
+	if (a->id < b->id)
+		return -1;
+	assert(a->id > b->id); // IDs cannot be equal
+	return 1;
+}
+
+struct wallet_payment **
 wallet_payment_list(const tal_t *ctx,
 		    struct wallet *wallet,
 		    const struct sha256 *payment_hash)
 {
-	const struct wallet_payment **payments;
+	struct wallet_payment **payments;
 	struct db_stmt *stmt;
 	struct wallet_payment *p;
 	size_t i;
 
-	payments = tal_arr(ctx, const struct wallet_payment *, 0);
+	payments = tal_arr(ctx, struct wallet_payment *, 0);
 	if (payment_hash) {
 		stmt =
 		    db_prepare_v2(wallet->db, SQL("SELECT"
```

Comment:
Any particular reason to use the asort algorithm instant of the ORDER BY SQL keyword?